tests/integration/test_schema_reader.py

```diff
@@ -0,0 +1,162 @@
+            "schema": json_encode(TRUE_SCHEMA.schema),
+        }
+        future = producer.send(
+            topic_name,
+            key=json_encode(key, binary=True),
+            value=json_encode(value, binary=True),
+        )
+        msg = future.get()
+
+        schema_reader.offset_watcher.wait_for_offset(msg.offset, timeout=5)
+
+        schemas = schema_reader.get_schemas(subject=subject, include_deleted=True)
+        assert len(schemas) == 1, "Deleted schemas must have been registered"
+
+        # Produce a soft deleted schema, this is the regression test
+        key = {
+            "subject": subject,
+            "version": 2,
+            "magic": 1,
+            "keytype": "SCHEMA",
+        }
+        test_global_schema_id = 2
+        value = {
+            "deleted": True,
+            "id": test_global_schema_id,
+            "subject": subject,
+            "version": 2,
+            "schema": json_encode(TRUE_SCHEMA.schema),
+        }
+        future = producer.send(
+            topic_name,
+            key=json_encode(key, binary=True),
+            value=json_encode(value, binary=True),
+        )
+        msg = future.get()
+
+        assert schema_reader.offset_watcher.wait_for_offset(msg.offset, timeout=5) is True
+        assert schema_reader.global_schema_id == test_global_schema_id
+
+        schemas = schema_reader.get_schemas(subject=subject, include_deleted=True)
+        assert len(schemas) == 2, "Deleted schemas must have been registered"
+
+
+def test_regression_config_for_inexisting_object_should_not_throw(
+    kafka_servers: KafkaServers,
+    producer: KafkaProducer,
+) -> None:
+    test_name = "test_regression_config_for_inexisting_object_should_not_throw"
+    subject = create_subject_name_factory(test_name)()
+    group_id = create_group_name_factory(test_name)()
+
+    config = set_config_defaults(
+        {
+            "bootstrap_uri": kafka_servers.bootstrap_servers,
+            "admin_metadata_max_age": 2,
+            "group_id": group_id,
+        }
+    )
+    master_coordinator = MasterCoordinator(config=config)
+    master_coordinator.start()
+    schema_reader = KafkaSchemaReader(config=config, master_coordinator=master_coordinator)
+    schema_reader.start()
+
+    with closing(master_coordinator), closing(schema_reader):
+        _wait_until_reader_is_ready_and_master(master_coordinator, schema_reader)
+
+        # Send an initial schema to initialize the subject in the reader, this is preparing the state
+        key = {
+            "subject": subject,
+            "magic": 0,
+            "keytype": "CONFIG",
+        }
+        value = ""  # Delete the config
+
+        future = producer.send(
+            DEFAULT_SCHEMA_TOPIC,
+            key=json_encode(key, binary=True),
+            value=json_encode(value, binary=True),
+        )
+        msg = future.get()
+
+        assert schema_reader.offset_watcher.wait_for_offset(msg.offset, timeout=5) is True
```

Comment:
This is a very fragile test, but for the current design of the schema reader it is the easiest I could do. Basically all exceptions are caught and the thread keeps running, so I couldn't do a `assert thread.is_alive()` because it never dies, so I used the offset_watcher instead, because when an exceptions happens the Event is not set.

IMO it would make sense to isolate the actual message handling code from the thread, so that can be tested directly, than this kinda of problem goes away (and the test setup is simpler)